src/main/java/com/pingcap/tikv/expression/scalar/ScalarFunction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv.expression.scalar;
+
+import com.pingcap.tidb.tipb.Expr;
+import com.pingcap.tidb.tipb.ExprType;
+import com.pingcap.tidb.tipb.FieldType;
+import com.pingcap.tidb.tipb.ScalarFuncSig;
+import com.pingcap.tikv.expression.TiExpr;
+import com.pingcap.tikv.exception.TiExpressionException;
+import com.pingcap.tikv.expression.TiFunctionExpression;
+import com.pingcap.tikv.types.DataType;
+import com.pingcap.tikv.util.ScalarFuncInfer;
+
+/**
+ * Scalar function
+ * Used in DAG mode
+ */
+public abstract class ScalarFunction extends TiFunctionExpression {
+  ScalarFunction(TiExpr... args) {
+    super(args);
+  }
+
+  /**
+   * Gets scalar function PB code representation.
+   *
+   * @return the pb code
+   */
+  ScalarFuncSig getSignature() {
+    return ScalarFuncInfer.of(
+        getArgTypeCode(),
+        getExprType()
+    );
+  }
+
+  private DataType getArgType() {
+    if (args.isEmpty()) {
+      throw new TiExpressionException(
+          "Scalar function's argument list cannot be empty!"
+      );
+    }
+
+    return args.get(0).getType();
```

Comment:
1. Do we assume types are coerced already here? 
2. What if it's a null type (typeless)
Say input is null < col (which is valid as expression but meaningless most of the time)

Reply:
1. We assume all the types are the same in argument list. If any of them is different from others, this expression may not be pushed down directly. We also assume `TiSpark` only create expressions with the same type in argument list.
2. Will throw an exception in `com.pingcap.tikv.predicates.RangeBuilder#exprToPoints`, we may need to open an issue later.

Reply:
Opened an [issue](https://github.com/pingcap/tikv-client-lib-java/issues/137).